cfgov/unprocessed/js/modules/Tree.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+/**
+ * Tree
+ * @class
+ *
+ * @classdesc A tree data structure.
+ * Tree's have one root node, and child nodes that branch.
+ * Like:
+ *
+ *        R
+ *       / \
+ *      A   B
+ *    / | \
+ *   C  D  E
+ *
+ * @returns {Tree} An instance.
+ */
+function Tree() {
+
+  var _root = null;
+  var _levelCache = {};
+
+  /**
+   * @param {Object} payload - Data to attach to the root node.
+   * @returns {Tree} An instance.
+   */
+  function init( payload ) {
+    _root = new TreeNode( this, payload );
+    _levelCache[0] = [ _root ];
+
+    return this;
+  }
+
+  /**
+   * @param {Object} payload - Data to attach to new node.
+   * @param {TreeNode} parent - Node on which to add a new child node.
+   * @returns {TreeNode} The newly instantiated and added node.
+   */
+  function add( payload, parent ) {
```

Comment:
Instead of payload, can we call it `data`? I think payload has a different connotation within the computing.